src/type-checker.cc

```diff
@@ -595,90 +595,62 @@ Result TypeChecker::OnTernary(Opcode opcode) {
   return CheckOpcode3(opcode);
 }
 
-Result TypeChecker::OnSimdLaneOp(Opcode opcode, uint64_t lane_idx) {
```

Comment:
@binji 
The reason I use uint64_t for the lane_idx here is:
ParseNat() will return an uint64_t number for lane_idx. It will leave out some parser errors if the lane_idx is truncated into int type. 

For example: 
For i8x16_extract_lane_s 0x1234567800000002, It should report an "lane index out for range" parser error, but it becomes validate if the 0x1234567800000002 is truncated to 0x00000002.

Reply:
Ah, I see. Thanks! I added a test for this.